### PR TITLE
fix(boot.go,src/healthsrv): add readiness and liveness probes

### LINKED ISSUE #54: Add k8s health and liveness checks
Need to add health and liveness checks!

### PR DESCRIPTION
**Pull request by arschles**
Fixes #54

After this is merged and a new container is built, https://github.com/deis/charts/issues/86 needs to be done

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -229,7 +226,7 @@ imports:
 - name: github.com/feyeleanor/slices
   version: bb44bb2e4817fe71ba7082d351fd582e7d40e3ea
 - name: github.com/fsouza/go-dockerclient
-  version: 76fd6c68cf24c48ee6a2b25def997182a29f940e
+  version: ""
```
Comment (bacongobbler) [inline glide.lock]: ?

File: manifests/deis-minio-rc.yaml [hunk 2]
```diff
@@ -15,11 +15,27 @@ spec:
...
       containers:
-        - imagePullPolicy: Always
-          name: deis-minio
-          image: quay.io/deisci/minio:v2-beta
+        - name: deis-minio
+          image: quay.io/arschles/minio:devel
+          imagePullPolicy: Always
+          env:
+            - name: HEALTH_SERVER_PORT
+              value: "8095"
```
Comment (bacongobbler) [inline manifests/deis-minio-rc.yaml]: is there a specific reason this is 8095? Why not just use the default 8082 port or change the default to 8095 instead? :)

File: glide.lock (hunk 1 above)
Comment (arschles) [inline glide.lock]: I'm not sure why glide did this. Investigating...

File: manifests/deis-minio-rc.yaml (hunk 2 above)
Comment (arschles) [inline manifests/deis-minio-rc.yaml]: Ah, thanks. I want to be explicit here, even if the specified port specified is the default. I'll fix the mismatch though!

**Comment by krancour** [comment]
I'm torn on whether it might be better to create the minio client used by the health check in / closer to the healthcheck code instead of doing so in `boot.go` and passing it in.  On the one hand, the complexity of creating that client seems like it should be handled a little closer to where it's used rather than the boot script which really handles overall orchestration.  On the other hand, secrets are used in constructing the minio client and you probably don't want to moves dependencies on kubernetes down into the healthcheck code.  Given that I can see both sides of this, I wouldn't let this stand in the way of my LGTM, but I just figured I'd mention it in case it inspires you in some way.

**Comment by arschles** [comment]
@krancour good point, and thanks for that. I think the other option for minio client creation location is in the `Serve` function in the `./src/healthsrv` package. I don't think it would harm anything, but it seemed like it was more prudent to create it inside `boot.go`, since it may be used elsewhere in the future. Also, it may be beneficial for tests, in the case where I may want to test the `Serve` function in the future (I don't right now though since it's very basic)

File: glide.lock (hunk 1 above)
Comment (arschles) [inline glide.lock]: @bacongobbler after running `glide up` again (more precisely, `make glideup`), this version number didn't come back. In https://github.com/arschles/deis-minio/commit/08099c9503ea2908b00df85b422fd4eeee27dab8, however, another version number did come back for a launchpad.net dependency...

@technosophos suggested that I file an issue on glide for this, so I'm working on getting a solid set of repro steps down before I do.

**Comment by mboersma** [comment]
This code looks good to me once the `glide` issue with go-dockerclient is sorted out.

**Comment by arschles** [comment]
@mboersma @bacongobbler after running a `glide up`, it looks like there are no more empty `version`s for github projects left

File: glide.lock [hunk 3]
```diff
@@ -1,44 +1,37 @@
...
-  subpackages:
-  - bigquery/v2
-  - googleapi
+  version: ""
+  repo: https://code.google.com/p/google-api-go-client
 - name: code.google.com/p/goprotobuf
-  version: 68415e7123da32b07eab49c96d2c4d6158360e9b
-  repo: https://github.com/golang/protobuf
+  version: ""
+  repo: https://code.google.com/p/goprotobuf
```
Comment (bacongobbler) [inline glide.lock:26]: code.google.com no longer exists. Any idea how this would occur?

File: glide.lock [hunk 4]
```diff
@@ -1,44 +1,37 @@
...
 imports:
 - name: bitbucket.org/bertimus9/systemstat
   version: 1468fd0db20598383c9393cccaa547de6ad99e5e
 - name: bitbucket.org/ww/goautoneg
   version: 75cd24fc2f2c
-- name: code.google.com/p/go-charset
-  version: ebbeafdc430e
 - name: code.google.com/p/go-uuid
-  version: 7dda39b2e7d5
+  version: ""
```
Comment (bacongobbler) [inline glide.lock:9]: looks like github refs are fixed, but not for any references to code.google.com

File: glide.lock (hunk 3 above)
Comment (arschles) [inline glide.lock:26]: no idea. glide seems to toggle that just about every time I run `glide up`

File: glide.lock (hunk 4 above)
Comment (arschles) [inline glide.lock:9]: yup, but that's what I'd expect. did I misunderstand?

Comment (bacongobbler) [inline glide.lock:9]: No I just realized this is the lockfile. Carry on :+1:

Comment (arschles) [inline glide.lock:9]: k, thanks for looking again

**Pull request merged**